lessons/12_scope/index.md

```diff
@@ -0,0 +1,148 @@
+# Lecture 12 Variable Scope
+
+Welcome to lecture 12 on variable scope in JavaScript
+
+## Midterm Announcement
+
+We have a midterm test coming up on Monday May 18th, 2020.
+
+- 30 min
+- conducted during video call
+- format:
+    - multiple choice
+    - short answer
+- will cover all we have learned thus far
+- topics:
+    - variables
+        - when to use `const` vs `let`
+        - scope
+    - arrays
+        - `.length`
+    - objects
+        - nested objects
+    - loops
+        - `for`
+        - `.forEach`
+    - github
+    - numbers and arithmetic
+    - strings
+    - callbacks
+    - the DOM
+        - selecting dom elements
+        - event listeners
+
+## Homework Process Reminder
+
+We have noticed a few problems with the homework so here are a few reminders
+
+- please do the homework, self learning is the only way to learn programming
+- merged pull requests without approves don't count
+- in order to get the certificate and complete the course one must complete 80% of the homework. We
+  have several students with 0% currently.
+
+## Final Projects Announcement
+
+As many of you know the second half of the semester is all about each student doing a final
+project. These final projects will be interactive apps using everything we've learned and more.
+They will give us a chance to tailor curriculum to each to student and push them.
+
+### Rough online of the final project process
+
+1. Decide on topic / goal of your webapp **<---- this week**
+  - visual design
+  - tech outline
+1. Next week start development
+1. Semiweekly check-ins and personalized goals / assignments
+1. Short presentation of final project during REDI demo day
+  - If this event does not take place in person there will be an alternative presentation format such
+    as blog post
+
+### What makes a good final project idea
+
+- creative
+- solves a problem
+- shows something previously unseen
+- unique
+- is interactive
+
+### Sample ideas
```

Comment:
I would remove these examples as we discussed yesterday. Since we want them to think first and only after give our suggestions

Reply:
removed